api/client/src/main/java/org/projectnessie/client/http/impl/UriBuilder.java

```diff
@@ -41,12 +41,12 @@ public UriBuilder(URI baseUri) {
   }
 
   public UriBuilder path(String path) {
-    if (uri.length() > 0) {
-      uri.append('/');
-    }
     String trimmedPath = HttpUtils.checkNonNullTrim(path);
     HttpUtils.checkArgument(
         !trimmedPath.isEmpty(), "Path %s must be of length greater than 0", trimmedPath);
+    if (uri.length() > 0 && !trimmedPath.startsWith("/")) {
```

Comment:
Changes in this method are due to the need to avoid double slashes in the URI path part.